emcc.py

```diff
@@ -956,6 +956,14 @@ def get_clang_output_extension(state):
     return '.o'
 
 
+def exec_subprocess_and_exit(cmd):
+  if utils.WINDOWS:
```

Comment:
I don't think this is necessary - Python has `execv` on Windows too, it just can't load cmd into the same process, so it falls back to the slower path automatically.

Might be cleaner to just use the same API on both platforms.

Reply:
Oh I didnt know they had emulation.  Thats odd.

It looks like there are maybe some issues with the emulation though: https://bugs.python.org/issue19066.. those seems like fairly major issues, but I guess it would be easy enough to detect and our tests would fail I believe.

Reply:
Oh huh, if there are issues, it's probably not worth it. I thought it would be cleaner and easier to just use single API, but if it doesn't fall back to whatever  `subprocess` does, that's a moot point.

Reply:
Although that one is from 2014 and was closed, so maybe not a problem anymore?

Reply:
Yes, I have to land a few other changes before this one (e.g. #20886), but if the tests all pass with execv on windows then I'll go with that option.

Reply:
Looks like you left this in?

Reply:
yeah, I figured we can try and take it out after and see what CI says.  I guess I could have tried it right away..